athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsUtil.java

```diff
@@ -0,0 +1,114 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.GOOGLE_SERVICE_ACCOUNT_JSON_TEMP_FILE_NAME;
+import static java.util.Objects.requireNonNull;
+
+public class GcsUtil
+{
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private GcsUtil()
+    {
+    }
+
+    /**
+     * Retrieves the GCS credential JSON from the JSON (key/value pairs)
+     *
+     * @param secretString String from the Secrets Manager
+     * @return GCS credentials JSON
+     */
+    public static String getGcsCredentialJsonString(final String secretString, String gcsCredentialKeysEnvVar) throws IOException
+    {
+        String appCredentialsJsonString = null;
+        if (secretString != null) {
+            TypeReference<HashMap<String, String>> typeRef
+                    = new TypeReference<>()
+            {
+            };
+            ObjectMapper mapper = new ObjectMapper();
+            Map<String, String> secretKeys = mapper.readValue(secretString.getBytes(StandardCharsets.UTF_8), typeRef);
+            appCredentialsJsonString = secretKeys.get(System.getenv(gcsCredentialKeysEnvVar));
+        }
+        return requireNonNull(appCredentialsJsonString, "GCS credential was null using key "
+                + gcsCredentialKeysEnvVar
+                + " in the secret " + System.getenv(gcsCredentialKeysEnvVar));
+    }
+
+    /**
+     * Builds a string representation of an instance of {@link StorageSplit}
+     *
+     * @param split An instance of {@link StorageSplit}
+     * @return String representation of an instance of {@link StorageSplit}
+     * @throws JsonProcessingException If JSON processing error happens
+     */
+    public static synchronized String splitAsJson(StorageSplit split) throws JsonProcessingException
+    {
+        return objectMapper.writeValueAsString(split);
+    }
+
+    public static boolean isFieldTypeNull(Field field)
+    {
+        return field.getType() == null
+                || field.getType().equals(Types.MinorType.NULL.getType());
+    }
+
+    public static void installCaCertificate() throws IOException
+    {
+        ClassLoader classLoader = GcsRecordHandler.class.getClassLoader();
+        File file = new File(requireNonNull(classLoader.getResource("")).getFile());
+        File src = new File(file.getAbsolutePath() + File.separator + "cacert.pem");
+        File dest = new File(Paths.get("/tmp").toAbsolutePath() + File.separator  + "cacert.pem");
+        if (!dest.exists()) {
+            Files.copy(src.toPath(), dest.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        }
+    }
+
+    public static void installGoogleCredentialsJsonFile(String gcsCredentialJsonString) throws IOException
+    {
+        File dest = new File(Paths.get("/tmp").toAbsolutePath() + File.separator  + GOOGLE_SERVICE_ACCOUNT_JSON_TEMP_FILE_NAME);
```

Comment:
Just use the `GOOGLE_APPLICATION_CREDENTIALS` environment variable as dest rather than hardcoding it to `/tmp/service-account.json`

This way you don't have to worry as much about the user changing this environment variable. As long as they set it to a path that we have write permissions to, it will just work and we don't have to worry about it.

However, you should probably make sure to create all the parent directories for the file that the environment variable points to (if the user sets something with more than one level then we'll fail even though the code could have just automatically created the parent paths).